client/app/pages/dashboards/public-dashboard-page.js

```diff
@@ -44,6 +44,8 @@ const PublicDashboardPage = {
             if (!isSafe) {
               error.errorMessage = 'This query contains potentially unsafe parameters and cannot be executed on a publicly shared dashboard.';
             }
+
+            throw error;
```

Comment:
Makes sense but does it have any affect on the flow?

Reply:
It doesn't right now, but as a good practice it's best to re-throw. (You never know if someone would want to handle this in the future)

Reply:
This won't blow up the dashboard load?

Reply:
Nope - gets caught by original promise.
https://github.com/getredash/redash/blob/c3cc65a21d74408da490ee076ae34bfd0391f5f6/client/app/services/widget.js#L139-L142